scalapb/scalapb_2.13/src/main/scala/com/linecorp/armeria/server/scalapb/ScalaPbNamedTypeInfoProvider.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.scalapb
+
+import com.linecorp.armeria.common.annotation.UnstableApi
+import com.linecorp.armeria.server.protobuf.ProtobufNamedTypeInfoProvider
+import com.linecorp.armeria.server.docs.{NamedTypeInfo, NamedTypeInfoProvider}
+import com.linecorp.armeria.server.scalapb.ScalaPbConverterUtil.isProtobufMessage
+
+/**
+ * A `NamedTypeInfoProvider` to create a `NamedTypeInfo` from a ScalaPB `GeneratedMessage`.
+ */
+@UnstableApi
+final class ScalaPbNamedTypeInfoProvider extends NamedTypeInfoProvider {
+  override def newNamedTypeInfo(typeDescriptor: Any): NamedTypeInfo = {
+    typeDescriptor match {
+      case clazz: Class[_] if isProtobufMessage(clazz) =>
+        val message = ScalaPbRequestConverterFunction.getDefaultInstance(clazz)
+        ProtobufNamedTypeInfoProvider.newStructInfo(message.companion.javaDescriptor).withAlias(clazz.getName())
```

Comment:
Q) I'm failing to understand the need for this `NamedTypeInfoProvider` 😅 

As far as I understand, the dependency `:scalapb_2.13` implies inclusion of `:protobuf`, which means both `ScalaPbNamedTypeInfoProvider` and `ProtobufNamedTypeInfoProvider` are included as `DocService#spiNamedTypeInfoProviders`.

Since it's undefined which `NamedTypeInfoProvider` will be applied first (and they seem to be essentially doing the same thing), I was wondering why we define a separate `NamedTypeInfoProvider` here.

Reply:
`ScalaPbNamedTypeInfoProvider` extracts protobuf `Descriptor` from `GeneratedMessage` and `GeneratedSealedOneof` using reflections and delegate it to `ProtobufNamedTypeInfoProvider`.
If `GeneratedMessage` is given to `ProtobufNamedTypeInfoProvider`, `null` is returned by `ProtobufNamedTypeInfoProvider`.

Reply:
> If GeneratedMessage is given to ProtobufNamedTypeInfoProvider, null is returned by ProtobufNamedTypeInfoProvider.

Sorry 😅    I wasn't able to understand this point.  Would you mind pointing out where `GeneratedMessage` yields `null`?

https://github.com/line/armeria/blob/5b3f3f1ce2663d3f34326f1725e6cb836c7fc73e/protobuf/src/main/java/com/linecorp/armeria/server/protobuf/ProtobufNamedTypeInfoProvider.java#L87-L109

Reply:
`scalapb.GeneratedMessage` is not a subclass of `com.google.protobuf.Message` so `isProtobufMessage()` returns false for `scalapb.GeneratedMessage`.
https://github.com/line/armeria/blob/11ee1ff1c5e830a3c52e1e09e452618357fbe2dd/protobuf/src/main/java/com/linecorp/armeria/server/protobuf/ProtobufNamedTypeInfoProvider.java#L102-L104

Reply:
Added some tests [`61dd8c0` (#4322)](https://github.com/line/armeria/pull/4322/commits/61dd8c0a5669f97d8cc3f5ad0c8e28558fedb2e9) in order to check your concerns. :-)